wasm/client.go

```diff
@@ -0,0 +1,242 @@
+	if !transitRelayAddress.Truthy() {
+		config.Set("transitRelayAddress", DEFAULT_TRANSIT_RELAY_ADDRESS)
+	}
+	if !passPhraseComponentLength.Truthy() {
+		config.Set("passPhraseComponentLength", DEFAULT_PASSPHRASE_COMPONENT_LENGTH)
+	}
+
+	// read config with defaults merged
+	appID = config.Get("appID")
+	rendezvousURL = config.Get("rendezvousURL")
+	transitRelayAddress = config.Get("transitRelayAddress")
+	passPhraseComponentLength = config.Get("passPhraseComponentLength")
+
+	client := &wormhole.Client{
+		AppID:                     appID.String(),
+		RendezvousURL:             rendezvousURL.String(),
+		TransitRelayAddress:       transitRelayAddress.String(),
+		PassPhraseComponentLength: passPhraseComponentLength.Int(),
+	}
+	clientPtr := uintptr(unsafe.Pointer(client))
+	clientMap[clientPtr] = client
+
+	return clientPtr
+}
+
+func Client_SendText(this js.Value, args []js.Value) interface{} {
+	ctx := context.Background()
+	fmt.Printf("this: %v\n", this)
+
+	return NewPromise(func(resolve ResolveFn, reject RejectFn) {
+		if len(args) != 2 {
+			reject(fmt.Errorf("invalid number of arguments: %d. expected: %d", len(args), 2))
+			return
+		}
+
+		clientPtr := uintptr(args[0].Int())
+		msg := args[1].String()
+		err, client := getClient(clientPtr)
+		if err != nil {
+			reject(err)
+			return
+		}
+
+		go func() {
+			code, _, err := client.SendText(ctx, msg)
+			if err != nil {
+				reject(err)
+				return
+			}
+			resolve(code)
+		}()
+	})
+}
+
+func Client_SendFile(this js.Value, args []js.Value) interface{} {
+	ctx := context.Background()
+	fmt.Printf("this: %v\n", this)
+
+	return NewPromise(func(resolve ResolveFn, reject RejectFn) {
+		if len(args) != 3 {
+			reject(fmt.Errorf("invalid number of arguments: %d. expected: %d", len(args), 3))
+			return
+		}
+
+		clientPtr := uintptr(args[0].Int())
+		fileName := args[1].String()
+
+		// TODO: something better!
+		fileData := make([]byte, MAX_FILE_SIZE)
+		js.CopyBytesToGo(fileData, args[2])
+		fileReader := bytes.NewReader(fileData)
+
+		err, client := getClient(clientPtr)
+		if err != nil {
+			reject(err)
+			return
+		}
+
+		go func() {
+			code, _, err := client.SendFile(ctx, fileName, fileReader)
+			if err != nil {
+				reject(err)
+				return
+			}
+			resolve(code)
+		}()
+	})
+}
+
+func Client_RecvText(this js.Value, args []js.Value) interface{} {
+	ctx := context.Background()
+
+	return NewPromise(func(resolve ResolveFn, reject RejectFn) {
+		if len(args) != 2 {
+			reject(fmt.Errorf("invalid number of arguments: %d. expected: %d", len(args), 2))
+			return
+		}
+
+		clientPtr := uintptr(args[0].Int())
+		code := args[1].String()
+		err, client := getClient(clientPtr)
+		if err != nil {
+			reject(err)
+			return
+		}
+
+		go func() {
+			msg, err := client.Receive(ctx, code)
+			if err != nil {
+				reject(err)
+				return
+			}
+
+			msgBytes, err := ioutil.ReadAll(msg)
+			if err != nil {
+				reject(err)
+				return
+			}
+			resolve(string(msgBytes))
+		}()
+	}).JSValue()
```

Comment:
I'm not sure if we need to do this or if it may be called implicitly otherwise. I seem to have forgotten it in other places and things seem to work as expected.